js/server/modules/@arangodb/foxx/manager.js

```diff
@@ -78,6 +78,26 @@ function isFoxxmaster () {
   return global.ArangoServerState.isFoxxmaster();
 }
 
+function validateInstallUrl (url) {
+  if (!internal.foxxAllowInstallFromRemote()) {
+    let checkRegex = /^https?:\/\/([^:\.]+:[^@\.]*@)?(www\.)?github\.com\//i;
+    // check if a user-defined install baseurl exists
+    let baseUrl = require('process').env.FOXX_BASE_URL;
+    if (baseUrl) {
+      checkRegex = new RegExp('^' + baseUrl.replace(/([\.+*\[\]\(\)\\])/g, '\\$1'));
```

Comment:
No, here we are escaping characters will special meanings, e.g. `.`, `+`, `*`, `[`, `]`, `(`, `)`, `\` with an additional backslash, so that we can use them safely in a RegExp.

Reply:
yeah but the resulting regexp is only used to determine if `url.startsWith(baseUrl)` in the case `FOXX_BASE_URL` is set, right? If that's the case I would prefer the more explicit check without the regexp.